src/app/reporting/Engine.cpp

```diff
@@ -177,6 +178,12 @@ DataModel::ActionReturnStatus RetrieveClusterData(DataModel::Provider * dataMode
     {
         status = *access_status;
     }
+    else if (IsSupportedGlobalAttributeNotInMetadata(readRequest.path.mAttributeId))
```

Comment:
I have to admit to being a little lost at this point: what guarantees that the cluster identified by `readRequest.path` exists here?  That should probably be documented...

Reply:
Nothing, however the read of global attributes validates cluster paths (because without a valid path it cannot receive metadata).

We should have path validity validation somewhere higher here, however this is what I inherited. I would prefer not to put too many changes into one PR.

Reply:
I also looked into adding validation in case it is a simple change - it is not, it has sideffects on tests. Did not fully check it, I believe it is because of wildcard expansion rules. In general since this is a 430 lines PR I think additional changes should be as followups. This generally keeps what we had before, with the exception that global attributes are pulled higher.

Reply:
If we know something is off here and will be completed in a follow up PR should we at the very least add a comment

Reply:
Added #37410 and comment referencing it.

Reply:
I'm not following the above discussion.

If we have nothing ensuring cluster actually exists here, is PR this not introducing a bug that we currently do not have?  Why is that OK?

Reply:
I thought it just moves what we had before higher up - we used to have `ReadAttribute` do the "if global, send it to global read" and Engine was doing ReadAttribute.

Now the update is that Engine does `if global, go global stuff otherwise readattribute` hence me thinking it does not introduce behavior changes. Granted I did rely that CI just passes and that has coverage.

From the slack discussion we do not have the "try reading a global attribute on a bad cluster/endpoint ... so I will add that".

Reply:
https://github.com/project-chip/connectedhomeip/pull/37345/commits/6495f1cebd00d086d6da61e905ebe15521b0d2b5 added a unit test to ensure no regression in the "write to read only attribute in various scenarios".

Hoping this is ok to prove no regression (I used both the list/global attributes and cluster revision as an example of "with metadata" and "without metadata" tests)

Reply:
OK, so I finally took the time to track down why this is OK.  The answer is that ReadGlobalAttributeFromMetadata expects the provider to fail out for the unsupported cluster/endpoint case and then does ValidateClusterPath to produce the right status.